### PR TITLE
[broker] Fix compatibility issue with other metadata store implementations in resources

### PR DESCRIPTION
**Pull request by Jason918**
### Motivation

There are some direct usages of zookeeper in brokers, these should be fixed as we supports multi metadata store implementations now.

### Modifications

Add a common `deleteIgnoreNotFoundAsync` in `BaseResources` which will not throw exception while deleting a non-exists path.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is already covered by existing tests, such as *(please describe tests)*.

### Does this pull request potentially affect one of the following parts:
- org.apache.pulsar.broker.admin.AdminTest#properties
- org.apache.pulsar.websocket.proxy.ProxyAuthorizationTest#test
- org.apache.pulsar.client.api.AuthenticatedProducerConsumerTest#testDeleteAuthenticationPoliciesOfTopic
- ...

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

Check the box below and label this PR (if you have committer privilege).

Need to update docs? 
  
- [x] `no-need-doc` 
Bug fix.

### CODE REVIEW

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java [hunk 1]
```diff
@@ -152,6 +152,20 @@ protected void delete(String path) throws MetadataStoreException {
         return cache.delete(path);
     }
 
+    protected CompletableFuture<Void> deleteIgnoreNotFoundAsync(String path) {
```
Comment (codelipenghui) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java:155]: How about deleteIfExists(String path)

**Review by codelipenghui — APPROVED** [review]

File: pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java [hunk 2]
```diff
@@ -152,6 +152,20 @@ protected void delete(String path) throws MetadataStoreException {
         return cache.delete(path);
     }
 
+    protected CompletableFuture<Void> deleteIfExistsAsync(String path) {
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        cache.delete(path).whenComplete((ignore, ex) -> {
```
Comment (merlimat) [inline pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java:157]: 👍 We could also do a check for `exists()` before the delete.

Comment (Jason918) [reply pulsar-broker-common/src/main/java/org/apache/pulsar/broker/resources/BaseResources.java]: Updated

**Review by merlimat — APPROVED** [review]

**Pull request merged**